### ISSUE TITLE
Question about the training procedure and the result

### ISSUE THREAD

**Issue by LSerene** [lolrudy/RBP_Pose#5]
Thank you for the code sharing and the impressive work. 

I followed the guidance and trained a model on the REAL275 dataset. However, the results seem not looking good. Since the instruction is quite simple, I am wondering if anything needs to be adjusted for the training procedure?

The following is the comparison of the results:
```
* The trained results:
  average mAP:
  3D IoU at 75: 66.4
  5 degree, 2cm: 34.1
  5 degree, 5cm: 43.6
  10 degree, 2cm: 58.3
  10 degree, 5cm: 76.1

* The results of the paper:
  3D IoU at 75: 67.8
  5 degree, 2cm: 38.2
  5 degree, 5cm: 48.1
  10 degree, 2cm: 63.1
  10 degree, 5cm: 79.2
```

Thank you so much!

**Comment by lolrudy** [comment]
Sorry for waiting. I find some minor differences in the code and modify them, now it should yield similar results. Moreover, you can extend the training epochs to 200 or 300 to ensure convergence.

**Comment by lolrudy** [comment]
FYI, I upload the trained model weights on NOCS-REAL275. Wondering if this could help you.

**Comment by luzzou** [comment]
> FYI, I upload the trained model weights on NOCS-REAL275. Wondering if this could help you.

Hi, I have tested the pretrained model on the REAL275 test set, but the results are 3-4 points lower than the results reported in the paper. I did not change the configurations, and I don't kown why.
Can you give me some advice?

Btw, the updated configuration file seems to be the same as the file uploaded 4 months ago.
![image](https://user-images.githubusercontent.com/29935938/202983865-bba45ce6-e377-44bd-99ce-21e2a06b1a9e.png)

**Comment by lolrudy** [comment]
I check the code again and find out two things lead to the inferior results:
1. The previous uploaded checkpoint is the first stage results, I re-upload the model.
2. We follow SPD and use the same definition of the canonical pose of the category "mug". Therefore, we change the GT pose in DualPoseNet results correspondingly. You can find the updated files in readme.

**Comment by luzzou** [comment]
> I check the code again and find out two things lead to the inferior results:
> 
> 1. The previous uploaded checkpoint is the first stage results, I re-upload the model.
> 2. We follow SPD and use the same definition of the canonical pose of the category "mug". Therefore, we change the GT pose in DualPoseNet results correspondingly. You can find the updated files in readme.

Thank you for your quick reply, I have re-evaluated the provided ckpt on the REAL275 dataset, now the results are very close to those reported in the paper.

![image](https://user-images.githubusercontent.com/29935938/203085567-b682a4e3-4821-47b8-b2f8-30304a1e6f04.png)

**Comment by LSerene** [comment]
> 

Thanks for replying. I got similar results using the provided weights. However, I got lower results using the updated code. By the way, I added `save_path = None` before line 87 of `SelfSketchPoseNet.py` since it cannot find the `save_path` parameter. Not sure if it influences the results.

Line 87 of `SelfSketchPoseNet.py` :
```python
point_depth_error, point_nocs_error = get_point_depth_error(PC_nocs, PC, gt_R, gt_t, gt_s, model=model_point, save_path=save_path)
```

The following are the results:

```bash
* The results using the provided weights:
  average mAP:
  3D IoU at 75: 67.9
  5 degree, 2cm: 38.2
  5 degree, 5cm: 48.1
  10 degree, 2cm: 63.1
  10 degree, 5cm: 79.3

* The new trained results:
  average mAP:
  3D IoU at 75: 66.4
  5 degree, 2cm: 31.7
  5 degree, 5cm: 41.1
  10 degree, 2cm: 57.3
  10 degree, 5cm: 75.4
```

**Comment by LSerene** [comment]
Close this issue as the provided weights is enough for my usage now.

**Comment by DHuiTnut** [comment]
> Close this issue as the provided weights is enough for my usage now.

Hi, sorry to bother you. Could you tell me how to generate '/Real/train/mug_handle.pkl' and 'results/mean_shape/mean_points_emb.npy'? Thanks a lot!!
